### DISCUSSION TITLE
TypeError: emit() got an unexpected keyword argument 'broadcast'

### DISCUSSION THREAD

**Discussion by Miniontoby** [miguelgrinberg/Flask-SocketIO#1972] [Q&A] [answered]
Hey guys,

With Version: `5.3.2`
I get this error: `TypeError: emit() got an unexpected keyword argument 'broadcast'`

Code that could reproduce the error:
```py
from flask import Flask
from flask_socketio import SocketIO
from apscheduler.schedulers.background import BackgroundScheduler
from apscheduler.executors.pool import ThreadPoolExecutor, ProcessPoolExecutor

app = Flask(__name__, static_url_path='/static')
socketio = SocketIO(app)

executors = {'default': ThreadPoolExecutor(16), 'processpool': ProcessPoolExecutor(4)}
sched = BackgroundScheduler(timezone='Asia/Seoul', executors=executors)

sensordata = {"Temperature": 0, "Humidity": 0, "Pressure": 0}

def getsensordata():
    global sensordata
    # I do get the data in a different way, but just to have a working code for you to test
    sensordata = {"Temperature":round(20.12, 2) , "Humidity":round(50.10, 2) , "Pressure":round(1012.12, 2)}
    socketio.emit('updateSensorData', sensordata, broadcast=True)

sched.add_job(getsensordata, 'interval', minutes=2)

if __name__ == "__main__":
    sched.start()
    socketio.run(app, host='0.0.0.0', port=5000)
    sched.shutdown()
```

Full Error message:
```py
Job "getsensordata (trigger: date[2023-04-15 10:58:16 CEST], next run at: 2023-04-15 10:58:16 CEST)" raised an exception
Traceback (most recent call last):
  File "/home/pi/flasks/flask_env/lib/python3.7/site-packages/apscheduler/executors/base.py", line 125, in run_job
    retval = job.func(*job.args, **job.kwargs)
  File "app.py", line 364, in getsensordata
    socketio.emit('updateSensorData', sensordata, broadcast=True)
  File "/home/pi/flasks/flask_env/lib/python3.7/site-packages/flask_socketio/__init__.py", line 463, in emit
    skip_sid=skip_sid, callback=callback, **kwargs)
TypeError: emit() got an unexpected keyword argument 'broadcast'
```

Also to note: I didn't make the original code! I am just using and improving it! So yeah be sure to keep that in mind

**Comment by miguelgrinberg** [comment]
The error tells you what's wrong. You are passing `broadcast` as an argument and this is invalid. See the docs: https://flask-socketio.readthedocs.io/en/latest/api.html#flask_socketio.SocketIO.emit

**Reply by Miniontoby** [reply]
But why did it work previously? I had no problems at all, but since it has updated and idk when... it doesn't work.

And I assume since this is not my code, that the owner also didn't have any problems/issues/errors with/on it

**Reply by miguelgrinberg** [reply]
I don't really know, since I don't know how the code was used. Maybe you used different versions of packages, and that made this work, even though the code is invalid. The code is (and always was) invalid, since `broadcast` is an argument that does not exist for the `socketio.emit()` method. Whoever wrote this may have confused `socketio.emit()` with the `emit()` standalone function, which has different behavior and does have a `broadcast` argument.

**Reply by Miniontoby** [reply]
The code can be seen at https://github.com/monkeymademe/piritos but yeah I opened an issue over there with the same content as here, but yeah idk if the code that there is might be doing some other things. 

Btw when using the `emit()` function from `from flask_socketio import emit` then it gives me `out of context` error or something

So yeah please take a look at the code and then tell me if what should actually be changed

**Reply by miguelgrinberg** [reply, marked as answer]
Try removing the bad argument.

**Reply by Miniontoby** [reply]
I did on my own runtime and that worked, but yeah idk if it might break the original code since the owner seemingly doesn't have any problems with the argument existing
